### PR TITLE
GTEST: Reduce the number of iterations (1k) for TCP

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Reduce the number of iterations (100k) for TCP

## Why ?

The tests (with 2kk iterations) takes too long

## How ?

Set the maximum number of iterations (100k) if has TCP transport

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9879/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9884/ for details (Mellanox internal link).

File: test/gtest/uct/test_uct_perf.cc [hunk 1]
```diff
@@ -164,7 +165,8 @@ UCS_TEST_P(test_uct_perf, envelope) {
             test->max *= UCT_PERF_TEST_MULTIPLIER;
             test->min /= UCT_PERF_TEST_MULTIPLIER;
         }
-        run_test(*test, 0, check_perf, GetParam()->tl_name, GetParam()->dev_name);
+        run_test(*test, max_iter, 0, check_perf,
```
Comment (yosefe) [inline test/gtest/uct/test_uct_perf.cc]: need to fix existing code+simplify this PR in following way:
1. remove "max_iter" param from run_multi_threaded() in this PR
2. create a copy of "*test" in here, before calling run_test(), and modify its min/max/max_iter fields (to fix pre-existing bug and pass max_iter)

Comment (dmitrygladkov) [reply test/gtest/uct/test_uct_perf.cc]: thanks! done

File: test/gtest/uct/test_uct_perf.cc [hunk 2]
```diff
@@ -153,18 +153,22 @@ UCS_TEST_P(test_uct_perf, envelope) {
 
     if (has_transport("tcp")) {
         check_perf = false; /* TODO calibrate expected performance based on transport */
+        max_iter   = 100000lu;
```
Comment (brminich) [inline test/gtest/uct/test_uct_perf.cc]: maybe reduce it even more (say 1000), since we do not check perf?

Comment (dmitrygladkov) [reply test/gtest/uct/test_uct_perf.cc]: UCT: 4095 ms -> 61 ms
UCP:  -> 1150 ms

I have no objections, @yosefe ?

then we can remove `max_iter` variable and do:
```
...
    if (!check_perf) {
        test.iters = 1000lu;
    }
    run_test(test, 0, check_perf, "", "");
}
...
```

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9890/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9892/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@yosefe could you review pls?

**Pull request merged**